issue_test.py

```diff
@@ -0,0 +1,47 @@
+from bauhaus import Encoding, proposition, constraint
+
+e = Encoding()
+
+@proposition(e)
+class V(object):
+    def __init__(self, val):
+        self.val = val
+
+    def __repr__(self):
```

Comment:
This should be defined by bauhaus as well, and the lack of _propname should indicate that it will be used (along with the other overloaded functions).

Reply:
I'd like to request clarification on the second half. If the user doesn't define prop_name are you saying Bauhaus should have a default implementation?

Reply:
No, the assertion was right. Just that you should indicate more what's going on with it (the fact that those other functions will be used).